### ISSUE TITLE
[AoT] scoped library not AoT compatible

### ISSUE THREAD

**Issue by matthiasscudlik** [tinesoft/generator-ngx-library#133]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```
Scoped libraries are not AoT compatible because of an invalid importAs in manifest.json.
Thus, neither the demo nor consuming applications can be compiled with AoT. 

### Generator Version
- 4.5.1

### Tools Version
yo 2.0.0
node 8.6.0
npm 5.3.0

### Targeted Angular version
4.x.x

### Repro steps.
- create a new library @foo/bar
- run gulp build 
- run gulp link
- cd demo
- npm install
- guilp build:demo

$ gulp build:demo
[11:47:54] Using gulpfile ~\git\ngx-scoped-lib\gulpfile.js
[11:47:54] Starting 'build:demo'...
Date: 2017-11-27T10:48:21.922Z
Hash: 744956cae4f037b1861b
Time: 20391ms
chunk {0} styles.f3d792d08f189686c411.bundle.css (styles) 159 kB {3} [initial] [rendered]
chunk {1} polyfills.3bc34265385d52184eab.bundle.js (polyfills) 86 bytes {3} [initial] [rendered]
chunk {2} main.e402deade8b026b7d50e.bundle.js (main) 84 bytes {3} [initial] [rendered]
chunk {3} inline.9d07561b59257af76b95.bundle.js (inline) 1.45 kB [entry] [rendered]

ERROR in C:/Users/myuser/git/ngx-scoped-lib/demo/src/$$_gendir/node_modules/@foo/bar/bar.ngfactory.ts (10,21): Cannot find module 'bar'.
ERROR in C:/Users/myuser/git/ngx-scoped-lib/demo/src/$$_gendir/app/home/home.component.ngfactory.ts (12,21): Cannot find module 'bar'.
ERROR in C:/Users/myuser/git/ngx-scoped-lib/demo/src/$$_gendir/app/home/home.module.ngfactory.ts (13,21): Cannot find module 'bar'.
ERROR in C:/Users/myuser/git/ngx-scoped-lib/demo/src/$$_gendir/app/app.module.ngfactory.ts (19,22): Cannot find module 'bar'.
ERROR in C:/Users/myuser/git/ngx-scoped-lib/demo/src/$$_gendir/app/app.server.module.ngfactory.ts (22,22): Cannot find module 'bar'.
[11:48:21] ng command failed. See below for errors.

[11:48:21] 1

### ROOT CAUSE:
- go to dist folder
- open `bar.metadata.json`
-> scroll to the end

Currently value of importAs
```json
"importAs":"bar"
```

but it should be

```json
"importAs":"@foo/bar"
```

=> the demo does not compile with AoT
=> apps using the library do not compile with AoT

### Desired functionality.
- importAs should contain the scope so that it is AoT compatible
- *please also port to 4.5.x branch*

**Comment by tinesoft** [comment]
Hi @matthiasscudlik 

Could you create a repro project? or give more info about what you've chaned for your lib.
'cause i cannot reproduce the issue with the standard lib generated by the tool...

Thanks

**Comment by matthiasscudlik** [comment]
@tinesoft 

heres the sample project, just generated it (with a different pc)
https://github.com/matthiasscudlik/bar

I also uploaded the dist folder, and as you can see the metadata.json contains importAs bar instead of @foo/bar:

https://github.com/matthiasscudlik/bar/blob/master/dist/bar.metadata.json

here is the complete output:
```shell
mas@MAS MINGW32 /e/git/bar (master)
$ gulp build
[19:38:54] Using gulpfile E:\git\bar\gulpfile.js
[19:38:54] Starting 'clean:dist'...
[19:38:54] Starting 'clean:coverage'...
[19:38:54] Starting 'clean:build'...
[19:38:54] Finished 'clean:dist' after 3.21 ms
[19:38:54] Finished 'clean:coverage' after 2.19 ms
[19:38:54] Finished 'clean:build' after 2.11 ms
[19:38:54] Starting 'clean'...
[19:38:54] Finished 'clean' after 18 μs
[19:38:54] Starting 'build'...
[19:38:54] Starting 'compile'...
[19:38:54] Starting 'lint'...
[19:38:56] Finished 'lint' after 1.7 s
[19:38:56] Starting 'pre-compile'...
[19:38:56] Finished 'pre-compile' after 16 ms
[19:38:56] Starting 'inline-templates'...
[19:38:56] Finished 'inline-templates' after 243 ms
[19:38:56] Starting 'ng-compile'...
[19:38:57] ES5 compilation succeeded.
[19:38:58] ES2015 compilation succeeded.
[19:38:58] Finished 'ng-compile' after 2.31 s
[19:38:58] Finished 'compile' after 4.26 s
[19:38:58] Starting 'test'...

START:

[at-loader] Using typescript@2.3.3 from typescript and "tsconfig.json" from E:\git\bar\src\tsconfig.spec.json.


[at-loader] Checking started in a separate process...

[at-loader] Ok, 0.008 sec.
27 11 2017 19:39:06.194:INFO [karma]: Karma v1.7.0 server started at http://0.0.0.0:9876/
27 11 2017 19:39:06.195:INFO [launcher]: Launching browser Chrome with unlimited concurrency
27 11 2017 19:39:06.243:INFO [launcher]: Starting browser Chrome
27 11 2017 19:39:07.975:INFO [Chrome 62.0.3202 (Windows 10 0.0.0)]: Connected on socket DWmXWXdOOvl9LF_TAAAA with id 86139286
  LibComponent
    ✔ should create component
    ✔ should have expected <p> text
  LibService
    ✔ should create service
    ✔ should say hello to stranger
    ✔ should say hello to provided user

Finished in 0.17 secs / 0.137 secs @ 19:39:08 GMT+0100 (Mitteleuropäische Zeit)

SUMMARY:
✔ 5 tests completed

=============================== Coverage summary ===============================
Statements   : 100% ( 13/13 )
Branches     : 100% ( 2/2 )
Functions    : 100% ( 5/5 )
Lines        : 100% ( 9/9 )
================================================================================
[19:39:08] Finished 'test' after 9.92 s
[19:39:08] Starting 'npm-package'...
[19:39:08] Finished 'npm-package' after 19 ms
[19:39:08] Starting 'rollup-bundle'...
[19:39:08] All bundles generated successfully.
[19:39:08] Finished 'rollup-bundle' after 140 ms
[19:39:08] Finished 'build' after 14 s

mas@MAS MINGW32 /e/git/bar (master)
$ gulp link
[19:39:17] Using gulpfile E:\git\bar\gulpfile.js
[19:39:17] Starting 'link'...
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN @foo/bar@0.0.1 requires a peer of @angular/core@^4.0.0 but none was installed.
npm WARN @foo/bar@0.0.1 requires a peer of @angular/common@^4.0.0 but none was installed.

up to date in 0.118s
C:\Users\mas\AppData\Roaming\npm\node_modules\@foo\bar -> E:\git\bar\dist
[19:39:19] Finished 'link' after 1.45 s

mas@MAS MINGW32 /e/git/bar (master)
$ cd demo

mas@MAS MINGW32 /e/git/bar/demo (master)
$ npm install

> node-sass@4.7.2 install E:\git\bar\demo\node_modules\node-sass
> node scripts/install.js

Downloading binary from https://github.com/sass/node-sass/releases/download/v4.7.2/win32-x64-57_binding.node
Download complete .] - :
Binary saved to E:\git\bar\demo\node_modules\node-sass\vendor\win32-x64-57\binding.node
Caching binary to C:\Users\mas\AppData\Roaming\npm-cache\node-sass\4.7.2\win32-x64-57_binding.node

> uglifyjs-webpack-plugin@0.4.6 postinstall E:\git\bar\demo\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js


> node-sass@4.7.2 postinstall E:\git\bar\demo\node_modules\node-sass
> node scripts/build.js

Binary found at E:\git\bar\demo\node_modules\node-sass\vendor\win32-x64-57\binding.node
Testing binary
Binary is fine

> @foo/bar-demo@0.0.0 postinstall E:\git\bar\demo
> npm link @foo/bar

E:\git\bar\demo\node_modules\@foo\bar -> C:\Users\mas\AppData\Roaming\npm\node_modules\@foo\bar -> E:\git\bar\dist
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN bootstrap@4.0.0-beta requires a peer of jquery@>=3.0.0 but none was installed.
npm WARN bootstrap@4.0.0-beta requires a peer of popper.js@^1.11.0 but none was installed.
npm WARN @schematics/angular@0.0.49 requires a peer of @angular-devkit/schematics@0.0.34 but none was installed.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 1109 packages in 56.221s

mas@MAS MINGW32 /e/git/bar/demo (master)
$ gulp build:demo
[19:40:31] Working directory changed to E:\git\bar
[19:40:34] Using gulpfile E:\git\bar\gulpfile.js
[19:40:34] Starting 'build:demo'...
Date: 2017-11-27T18:40:48.395Z
Hash: 7602f38a49c17ce35fcc
Time: 9568ms
chunk {0} styles.f3d792d08f189686c411.bundle.css (styles) 159 kB {3} [initial] [rendered]
chunk {1} polyfills.3bc34265385d52184eab.bundle.js (polyfills) 86 bytes {3} [initial] [rendered]
chunk {2} main.e402deade8b026b7d50e.bundle.js (main) 84 bytes {3} [initial] [rendered]
chunk {3} inline.9d07561b59257af76b95.bundle.js (inline) 1.45 kB [entry] [rendered]

ERROR in E:/git/bar/demo/src/$$_gendir/node_modules/@foo/bar/bar.ngfactory.ts (10,21): Cannot find module 'bar'.
ERROR in E:/git/bar/demo/src/$$_gendir/app/home/home.component.ngfactory.ts (12,21): Cannot find module 'bar'.
ERROR in E:/git/bar/demo/src/$$_gendir/app/home/home.module.ngfactory.ts (13,21): Cannot find module 'bar'.
ERROR in E:/git/bar/demo/src/$$_gendir/app/app.module.ngfactory.ts (19,22): Cannot find module 'bar'.
ERROR in E:/git/bar/demo/src/$$_gendir/app/app.server.module.ngfactory.ts (22,22): Cannot find module 'bar'.
[19:40:48] ng command failed. See below for errors.

[19:40:48] 1

mas@MAS MINGW32 /e/git/bar/demo (master)
```

**Comment by tinesoft** [comment]
Ok thanks for the repro step: 👍 

I'll look into into it right away.

**Comment by matthiasscudlik** [comment]
if you can reproduce the error you can manually change the metadata.json to see that gulp build:demo works afterwards. Unfortunately i didn't find where importAs is set

**Comment by tinesoft** [comment]
Ok, Just released [v5.3.0](https://github.com/tinesoft/generator-ngx-library/releases/tag/v5.3.0) with the fix.
I don't maintain intermediate version (too much work, and i want to encourage people to use newer versions of the generator). Anyway the fix is quite simple, see https://github.com/tinesoft/generator-ngx-library/commit/18de2fb171eabe6ffe55c1eab7597725cb2975f9. You just have to edit both `src/tsconfig.lib.json` and `src/tsconfig.lib.es5.json` as such:

```
  "angularCompilerOptions": {
...
    "flatModuleId": "@foo/bar", // previously was just "bar" which is wrong for scope libs
...
  }
```

Thanks for reporting and enjoy!

**Comment by matthiasscudlik** [comment]
works like a charm, thanks a lot! :)
